docs/source/testing/mocking-responses.mdx

```diff
@@ -0,0 +1,181 @@
+---
+title: Mocking responses
+---
+
+Apollo Kotlin provides two tools to mock responses in tests:
+
+- `TestNetworkTransport`: allows to specify the GraphQL responses returned by `ApolloClient`
+- `MockServer`: a low-level HTTP server that can be used to mock any responses
+
+In most cases, `TestNetworkTransport` will be more straightforward to use in your tests, while `MockServer` can be
+useful to test specific server behavior, such as error cases, custom headers, timeouts, etc.
+
+## `TestNetworkTransport`
+
+This is a high level test API which allows to specify the responses that will be returned by your `ApolloClient`.
+
+Add the dependency to your project's `build.gradle` file:
+
+```kotlin:title=build.gradle[.kts]
+dependencies {
+  testImplementation("com.apollographql.apollo3:apollo-testing-support:x.y.z")
+}
+```
+
+Enable the `TestNetworkTransport` by passing it to the `ApolloClient` builder:
+
+```kotlin{2}
+val apolloClient = ApolloClient.Builder()
+    .networkTransport(TestNetworkTransport())
+    .build()
+```
+
+Either keep a reference to the `TestNetworkTransport` instance, or use the `testNetworkTransport` extension function to
+retrieve it:
+
+```kotlin
+val testNetworkTransport = apolloClient.testNetworkTransport
+```
+
+You can then use the `register` function to specify the responses that will be returned in function of the received
+operations:
+
+```kotlin{7}
+val query = GetHeroQuery("001")
+val testData = GetHeroQuery.Data {
+  hero = droidHero {
+    name = "R2D2"
+  }
+}
+apolloClient.testNetworkTransport.register(query, testData)
+
+val actual = apolloClient.query(query).execute().data!!
+assertEquals(testData.hero.name, actual.hero.name)
+```
+
+The `register` function accepts an `Operation` (here, the query) and the `ApolloResponse` to be returned when it is
+executed. As a shortcut you can pass a `Data` directly instead of an `ApolloResponse`, like in the example above.
+
+Note how you can use the [Test builders](./test-builders/) to easily create the data.
```

Comment:
Life is not easy :P. 

Joking asides, "easy" is a very relative concept so different readers might have different expectations. I'd replace with "type-safe", which is objectively true and the same for everyone

```suggestion
Note how you can use the [Test builders](./test-builders/) to create the data in a type-safe way.
```

Reply:
Totally agree 🙏 Trying to avoid this kind of terms, but they slip 😅